contrib/custom-image-kubelet/genconfig.py

```diff
@@ -0,0 +1,217 @@
+                archive_url = base_url._replace(path=a['archive']['path']).geturl()
+                extract_network_tar(archive_url, bin_dir)
+
+
+def extract_network_tar(archive_url, bin_dir):
+    logging.info(f'Extracting files from {archive_url}')
+    with requests.get(archive_url, stream=True) as response, io.BytesIO() as buf:
+        response.raise_for_status()
+        copyfileobj(response.raw, buf)
+        buf.seek(0, 0)
+
+        with tarfile.open(fileobj=buf) as tar:
+            for member in tar.getmembers():
+                if member.isreg():
+                    member.name = basename(member.name)
+                    logging.info(f'Extracting {bin_dir}/{member.name}')
+                    tar.extract(member, bin_dir)
+
+
+def write_ecr_credentials(release, prefix):
+    etc_dir = f'{prefix}/etc/rancher/rke2'
+    registries = get_ecr_registries(release)
+    if not registries:
+        return
+
+    registry_configs = dict()
+    registry_regions = defaultdict(list)
+
+    for registry in registries:
+        parts = registry.split('.')
+        registry_regions[parts[3]].append(parts[0])
+
+    for region, registries in registry_regions.items():
+        logging.info(f'Getting auth tokens for {registries} in {region}')
+        response = boto3.client('ecr', region_name=region).get_authorization_token(registryIds=registries)
+        for auth in response.get('authorizationData', []):
+            endpoint = auth['proxyEndpoint'].split('//')[1]
+            username, password = b64decode(auth['authorizationToken']).decode().split(':')
+            registry_configs[endpoint] = {'auth': {'username': username, 'password': password}}
+
+    try:
+        os.makedirs(normpath(etc_dir), mode=0o0700)
+    except FileExistsError:
+        pass
+
+    with open(normpath(f'{etc_dir}/registries.yaml'), mode='a+') as registries_file:
+        registries_file.seek(0, 0)
+        registries_yaml = load(registries_file, Loader=Loader)
+        if not registries_yaml:
+            registries_yaml = dict()
+        if 'configs' not in registries_yaml:
+            registries_yaml['configs'] = dict()
+        registries_yaml['configs'].update(registry_configs)
+        logging.info(f'Writing credentials to {registries_file.name}')
+        registries_file.seek(0, 0)
+        registries_file.truncate()
+        dump(registries_yaml, registries_file, Dumper=Dumper)
+
+
+def get_ecr_registries(release):
+    registries = set()
+    for c in release.get('status', {}).get('components', []):
+        for a in c.get('assets', []):
+            if 'image' in a:
+                registry = a['image']['uri'].split('/')[0]
+                if '.ecr.' in registry:
+                    registries.add(registry)
+    return registries
+
+
+if __name__ == '__main__':
+    try:
+        logging.basicConfig(format="%(levelname).1s %(message)s", level=logging.INFO, stream=sys.stdout)
+        main()
+    except (KeyboardInterrupt, BrokenPipeError):
```

Comment:
We want to ignore ctrl-c?

Reply:
No, at this point the exception has already broken out of main(). This just makes it exit cleanly instead of printing an error.